c_tests/tests/simple_interp_loop.c

```diff
@@ -0,0 +1,79 @@
+// Compiler:
+//   env-var: YKD_PRINT_JITSTATE=1
+// Run-time:
+//   env-var: YKD_PRINT_IR=jit-pre-opt
+//   stderr:
+//     jit-state: start-tracing
+//     ...
+//     define internal %IOStruct @__yk_compiled_trace_0(%IOStruct %0) {
+//        ...
+//     }
+//     ...
+//     jit-state: stop-tracing
+//     0
+//     1
+//     2
+//     3
+//     jit-state: enter-jit-code
+//     0
+//     1
+//     2
+//     3
+//     jit-state: exit-jit-code
+//     ...
+#include <assert.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <yk.h>
+#include <yk_testing.h>
+
+// The sole mutable memory cell of the interpreter.
+int mem = 3;
+
+// The bytecodes accepted by the interpreter.
+#define NOP 0
+#define DEC 1
+#define RESTART_IF_NOT_ZERO 2
+#define EXIT 3
+
+int
+main(int argc, char **argv)
+{
+  // A hard-coded program to execute.
+  int prog[] = {0, 0, 1, 2, 0, 3};
```

Comment:
Do we need such a long "program"? I know that sounds silly, but my guess is that when we start doing optimisations and so on, weird things will break, and then we'll be grateful to have a test called `simple_interp_loop` be the simplest possible interpreter loop! Maybe, in fact, we should have a sequence of "gradually less simple" interpreter loops? e.g. one loop that just has a `DEC` instruction, one that has `RESTART_IF_NOT_ZERO`, and so on?